### PR TITLE
Configure VSCode to use hard tabs on TS and Svelte files

### PR DESCRIPTION
**Pull request by whitphx**
# Description

Using hard-tabs is [ensured by Prettier](https://github.com/gradio-app/gradio/blob/3227a9b703fb520bf4f8a8ee08f9027c0a5c1070/.config/.prettierrc.json#LL2C18-L2C18).
I want to configure VSCode too.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by pngwn** [comment]
You can install the prettier extension to do this. I thought we had some recommended extension for this repo. I can look into that.

**Comment by whitphx** [comment]
Thank you, but it does not work in my env... probably because of https://github.com/prettier/prettier-vscode/issues/1859.

**Comment by pngwn** [comment]
Ah I see. We have this set in both prettier and editorconfig, which I was hoping would make it go. No big deal, we can merge this in if it fixes things for you.

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ No changes to highlight.
 
 ## Full Changelog:
 
-No changes to highlight.
+* Updated `.vscode/settings.json` to use tabs in TS and Svelte files, by [@whitphx](https://github.com/whitphx) in [PR 4161](https://github.com/gradio-app/gradio/pull/4161)"
```
Comment (pngwn) [inline CHANGELOG.md:28]: We don't need a changelog entry for this either.

**Comment by whitphx** [comment]
Ah, OK, thanks. editorconfig may work. So I will withdraw this PR.

Instead, I think  it would be nice if `.vscode/extensions.json` is provided with at least the Prettier and EditorConfig plugins. What do you think?

**Comment by pngwn** [comment]
Yeah, definitely. I think the svelte and vscode extensions are the essential ones. And i'll be adding some linting soon so we can add eslint when that happens too.

**Pull request closed without merging**